dali/python/nvidia/dali/pipeline.py

```diff
@@ -434,6 +434,16 @@ def executor_statistics(self):
             raise RuntimeError("Pipeline must be built first.")
         return self._pipe.executor_statistics()
 
+    def external_source_shm_statistics(self):
```

Comment:
These are not really "statistics" - rather, "capacities". If we plan to add more than just the capacity, how about returning a dictionary / named tuple?
```
{
"capacity":[1000, 223487, 1000, 1000],
}
```
A NamedTuple would probably be nicer for the end-user.

A bare minimum is an array of arrays - so we can add more info just by extending the output array - I don't like it very much, since now it relies on documentation to get any semantics.

Reply:
I changed it to be a dicionary, very good point.